modal/functions.py

```diff
@@ -932,16 +932,20 @@ async def _load(self: _Function, resolver: Resolver, existing_object_id: Optiona
         obj._is_method = False
         obj._spec = function_spec  # needed for modal shell
 
-        # Used to check whether we should rebuild an image using run_function
-        # Plaintext source and arg definition for the function, so it's part of the image
-        # hash. We can't use the cloudpickle hash because it's not very stable.
+        # Used to check whether we should rebuild a modal.Image which uses `run_function`.
         gpus: List[GPU_T] = gpu if isinstance(gpu, list) else [gpu]
         obj._build_args = dict(  # See get_build_def
             secrets=repr(secrets),
             gpu_config=repr([parse_gpu_config(_gpu) for _gpu in gpus]),
             mounts=repr(mounts),
             network_file_systems=repr(network_file_systems),
         )
+        # these key are excluded if empty to avoid rebuilds on client upgrade
```

Comment:
Will this trigger rebuilds if _not_ empty?

Reply:
Yes, but that's unavoidable isn't it?

Reply:
I guess. Worth highlighting in a changelog note at least.